src/EmailAddress.php

```diff
@@ -473,7 +473,7 @@ protected function splitEmailParts($value)
         }
 
         $this->localPart = $matches[1];
-        $this->hostname  = $matches[2];
+        $this->hostname  = $this->idnToUtf8($matches[2]);
```

Comment:
hostname should host the ascii representation as is more portable for any other use of the value.

For example the following condition `(strlen($this->hostname) > 255)` is now bugged as strlen does not allow multibyte characters (UTF-8)